File: SkaiLab.Invoice.Models/CustomerTransaction.cs
using System;
using System.Collections.Generic;

namespace SkaiLab.Invoice.Models
{
    public class CustomerTransaction
    {
        public long Id { get; set; }
        public long CustomerId { get; set; }
        public string OrganisationId { get; set; }
        public int StatusId { get; set; }
        public string RefNo { get; set; }
        public string Note { get; set; }
        public DateTime Date { get; set; }
        public int CurrencyId { get; set; }
        public decimal TaxCurrencyExchangeRate { get; set; }
        public decimal BaseCurrencyExchangeRate { get; set; }
        public decimal Total { get; set; }
        public decimal TotalIncludeTax { get; set; }
        public bool IsTaxIncome { get; set; }
        public string CreatedBy { get; set; }
        public DateTime Created { get; set; }
        public string PaidBy { get; set; }
        public DateTime? PaidDate { get; set; }
        public DateTime? DueDate { get; set; }
        public string Number { get; set; }

        public virtual Currency Currency { get; set; }
        public virtual Customer Customer { get; set; }
        public string TermAndCondition { get; set; }
      
        public List<Attachment> Attachments { get; set; }
        public  List<CustomerTransactionItem> CustomerTransactionItems { get; set; }
        public Organsation Organisation { get; set; }
    }
    public  class CustomerTransactionItem
    {
        public long Id { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public double? DiscountRate { get; set; }
        public long? TaxId { get; set; }
        public decimal LineTotal { get; set; }
        public decimal LineTotalIncludeTax { get; set; }
        public string Description { get; set; }
        public long? ProductId { get; set; }
        public long? LocationId { get; set; }
        public  Location Location { get; set; }
        public  Product Product { get; set; }
        public Tax Tax { get; set; }
        public long CustomerTransaction { get; set; }

    }
    public class Invoice:CustomerTransaction
    {
        public InvoiceStatus Status { get; set; }
        public long? QuoteId { get; set; }
    }
    public class InvoiceStatus
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Count { get; set; }
       
    }
    public class CustomerCreditStatus
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Count { get; set; }

    }
    public class CustomerCredit : CustomerTransaction
    {
        public CustomerCreditStatus Status { get; set; }
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/customer-transaction.ts
import { Attachment } from "./attachment";
import { Currency } from "./currency";
import { Customer } from "./customer";
import { Product } from "./product";
import { Tax } from "./tax";

export class CustomerTransaction{
    id:number;
    customerId:number;
    organisationId:string;
    refNo:string;
    note:string;
    date:Date;
    currencyId:number;
    taxCurrencyExchangeRatefNo:number;
    baseCurrencyExchangeRate:number;
    total:number;
    totalIncludeTax:number;
    isTaxIncome:boolean;
    paidDate:Date;
    dueDate:Date;
    number:string;
    customer:Customer;
    attachments:Attachment[];
    currency:Currency;
    selected:boolean;
    termAndCondition:string;
    customerTransactionItems:CustomerTransactionItem[];
    constructor(){
        this.id=0;
        this.customerId=null;
        this.organisationId="";
        this.refNo="";
        this.note="";
        this.date=new Date();
        this.currencyId=0;
        this.taxCurrencyExchangeRatefNo=0;
        this.baseCurrencyExchangeRate=0;
        this.total=0;
        this.totalIncludeTax=0;
        this.isTaxIncome=false;
        this.paidDate=null;
        this.dueDate=new Date();
        this.number="";
        this.customer=new Customer();
        this.attachments=[];
        this.customerTransactionItems=[];
        this.currency=new Currency();
        this.selected=false;
        this.termAndCondition="";
    }

}
export class CustomerTransactionItem{
    id:number;
    productId:number;
    quantity:number;
    taxId?:number;
    lineTotal:number;
    lineTotalIncludeTax:number;
    product:Product;
    tax:Tax;
    discountRate?:number;
    unitPrice:number;
    locationId:number;
    description:string;
    inventoryQty:number;
    requestingInventory:boolean;
    
    constructor(){
        this.id=0;
        this.productId=null;
        this.product=new Product();
        this.quantity=null;
        this.taxId=null;
        this.discountRate=null;
        this.tax=null;
        this.lineTotal=0;
        this.lineTotalIncludeTax=0;
        this.unitPrice=0;
        this.locationId=null;
        this.description="";
        this.inventoryQty=null;
        this.requestingInventory=false;
        
    }
}
export class Invoice extends CustomerTransaction{
    statusId:number;
    status:InvoiceStatus;
    constructor(){
        super();
        this.statusId=0;
        this.status=new InvoiceStatus();
    }
}
export class InvoiceStatus{
    id:number;
    name:string;
    constructor(){
        this.id=0;
        this.name="";
    }
}
export class CustomerCreditStatus{
    id:number;
    name:string;
    constructor(){
        this.id=0;
        this.name="";
    }
}
export class CustomerCredit extends CustomerTransaction{
    statusId:number;
    status:InvoiceStatus;
    constructor(){
        super();
        this.statusId=0;
        this.status=new InvoiceStatus();
    }
}